components/log-ingestor/tests/test_ingestion_job.rs

```diff
@@ -67,27 +72,89 @@ async fn receive_object_metadata(
     received_objects
 }
 
+/// Uploads S3 objects and receives their metadata via the provided receiver channel.
+///
+/// Objects are created with keys formatted as `{prefix}/{i}.log` where `i` is the object index.
+///
+/// # Returns
+///
+/// A tuple containing:
+///
+/// * A vector of created S3 object metadata.
+/// * A vector of received S3 object metadata.
+async fn upload_and_receive(
+    s3_client: aws_sdk_s3::Client,
+    bucket: String,
+    prefix: String,
+    num_objects_to_create: usize,
+    receiver: mpsc::Receiver<ObjectMetadata>,
+) -> (Vec<ObjectMetadata>, Vec<ObjectMetadata>) {
+    let objects_to_create: Vec<_> = (0..num_objects_to_create)
+        .map(|idx| ObjectMetadata {
+            bucket: bucket.clone(),
+            key: format!("{prefix}/{idx:05}.log"),
+            size: 16,
+        })
+        .collect();
+
+    let creation_handler = tokio::spawn(create_s3_objects(
+        s3_client.clone(),
+        objects_to_create.clone(),
+    ));
+
+    let objects_received = tokio::time::timeout(
+        Duration::from_secs(RECEIVER_TIMEOUT_SEC),
+        receive_object_metadata(receiver, objects_to_create.len()),
+    )
+    .await
+    .expect("Timed out while receiving object metadata");
+
+    let () = creation_handler
```

Comment:
Do I miss something?
```suggestion
    creation_handler
```